plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/MavenUtils.java

```diff
@@ -177,4 +180,18 @@ public static ArtifactRepository createRepository(String id, String url) throws
     return MavenPlugin.getMaven().createArtifactRepository(id, url);
   }
 
+  public static boolean isArtifactAvailableLocally(String groupId, String artifactId, String version) {
+    try {
+      Preconditions.checkArgument(!MAVEN_LATEST_VERSION.equals(version));
+      String artifactPath =
+          MavenPlugin.getMaven().getLocalRepository()
+          .pathOf(new DefaultArtifact(groupId, artifactId, version, null /* scope */, "jar",
+                                      null /*classifier */, new DefaultArtifactHandler("jar")));
+      return new File(artifactPath).exists();
+    } catch (CoreException ex) {
+      logger.log(Level.SEVERE, "Could not lookup local repository", ex);
```

Comment:
Not sure what casually means in this case. I think not finding the local repo is rather serious error and fundamentally affects most M2E actions.